### PR TITLE
Remove uses of state_school_id

### PR DESCRIPTION
**Pull request by TurnerRiley**
## Overview
This PR removes uses of the `state_school_id` field in our code base. This will be followed up by a PR that actually removes that column from the `schools` data table.

## Context
There are 100+ schools that are getting skipped when ingesting NCES stats data from the 2021-2022 school year. Vijaya did some digging and found that it was because the school data itself wasn’t ingested for these schools because there was an existing entry with the same state school ID, but different NCES school ID. According to the current schema constraints, the rule says every entry should have a unique NCES school ID and a unique state school ID. So, anytime the NCES school ID alone changes, that entry is ignored as duplicate.

[Vijaya discussed more with Baker](https://codedotorg.slack.com/archives/C04540KNGDQ/p1682359022125929) and found that it appears the state school ID is repurposed, so we have instances where a completely new school gets a new NCES ID but uses the same state school ID as that of a school that is potentially discontinued. There are also a few instances where minor details about the school changes, during which the state school ID changes, but not the NCES ID.

The original plan was to remove the uniqueness constraint on the `state_school_id` column in the schools data table. This way, the NCES ID will still be unique and a way to differentiate between what we define as a "school", while avoiding any issues when it comes to the different ways states ID their schools. However, upon [further discussion](https://codedotorg.slack.com/archives/C04540KNGDQ/p1695228596440659) with Dave, Baker, and Hannah W., we decided that `state_school_id` was no longer needed.

## Demo regarding missing schools
To start, I just tried seeding schools from the 2021-2022 data set. The output stated that 118 schools were skipped (the only reason a skip would occur is a duplicate NCES ID or state ID):
118_skipped

After running the migration in this PR, the output stated `0 schools were skipped`:
0_skipped

I then ran `rake seed:schools` for seedings schools from all the data sets we pull from, and saw the same output for each file of `0 schools were skipped`.

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?modal=detail&selectedIssue=ACQ-972&assignee=60d62161f65054006980bd71)
Slack discussion about school seeding issue: [here](https://codedotorg.slack.com/archives/C04540KNGDQ/p1682359022125929)
Slack discussion about removing `state_school_id`: [here](https://codedotorg.slack.com/archives/C04540KNGDQ/p1695228596440659)

## Testing story
Local testing and updating unit tests.

## Follow-up work
PR to remove `state_school_id` from data tables.

## PR Checklist:
- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/db/schema.rb
Comment (TurnerRiley) [inline dashboard/db/schema.rb]: Should I update the schema here or in the PR with the actual database changes?

File: dashboard/app/models/school.rb
Comment (davidsbailey) [inline dashboard/app/models/school.rb]: please do not remove this line yet, it will be removed by the migration (and will cause minor issues if removed sooner)

File: dashboard/app/models/school.rb
Comment (davidsbailey) [inline dashboard/app/models/school.rb]: please also hold off on this line until the migration PR

File: dashboard/app/models/school.rb
Comment (davidsbailey) [inline dashboard/app/models/school.rb]: revert debug code?

File: dashboard/db/schema.rb
Comment (davidsbailey) [reply dashboard/db/schema.rb]: please wait, this will also be taken care of by the migration

**Review by davidsbailey — APPROVED** [review]
Thank you for the thorough explanation! LGTM after restoring a few lines as noted in the PR comments.

**Comment by davidsbailey** [comment]
I also noticed that this directory can be removed: https://github.com/code-dot-org/code-dot-org/tree/c9038bbc3ba109c6460cb594a2f8039458922015/dashboard/config/state_cs_offerings maybe in a follow up PR?

File: dashboard/app/models/school.rb
Comment (TurnerRiley) [reply dashboard/app/models/school.rb]: Oh yes thanks for catching that!

**Comment by TurnerRiley** [comment]
> I also noticed that this directory can be removed: https://github.com/code-dot-org/code-dot-org/tree/c9038bbc3ba109c6460cb594a2f8039458922015/dashboard/config/state_cs_offerings maybe in a follow up PR?

I'll make a ticket for this!

**Pull request merged**
